client/download.go

```diff
@@ -138,6 +147,35 @@ func downloadFile(httpClient httpClient, filepath string, url string, devnull bo
 		}
 	}
 
+	size, err = downloadFile(httpClient, temppath, url, expectedSha)
+	if err != nil {
+		return size, err
```

Comment:
We should remove the tempfile here IMHO.

Reply:
right, thanks